js/generator.js

```diff
@@ -749,7 +746,7 @@ var DungeonGenerator=function(mapwidth,mapheight,seed,debug) {
 
 		questsStructure.forEach(entry=>{
 			for (var i=0;i<entry.count;i++)
-				if (quest=this.addQuest(quests[entry.questType],addedQuests)) addedQuests.push(quest);
+				if (quest==this.addQuest(quests[entry.questType],addedQuests)) addedQuests.push(quest);
```

Comment:
I think this is the only possible breaking change. I'm assuming this was a typo and you wanted to do an equality check instead of an assign, which would always return true.

Reply:
Thank for the contribution!

That wasn't a typo: it's a way to both check a condition and cache a value.

`addQuest` will return the quest data if there is a route in the dungeon that satisfies its requirements or `false` if it wasn't possible. This line both check if the quest was added and stores it into the `quest` variable, so:

- If the quest was added the condition is satisfied, since objects are trueish,`quest` is set to the quest data, and `quest` is added to the `addedQuests` array, that's later passed to `addQuest`
- If the quest wasn't added, the condition is not satisfied, and quest contains a `false` value. The `addedQuests` array is not changed.